manufacturing-client/src/main.rs

```diff
@@ -937,3 +956,22 @@ impl KeyReference {
         }
     }
 }
+
+fn get_default_network_iface() -> Result<Option<String>, std::io::Error> {
+    let file = File::open("/proc/net/route")?;
+    let reader = BufReader::new(file);
+
+    for line in reader.lines().skip(1) {
+        let line = line.unwrap();
+        let fields: Vec<_> = line.split_whitespace().collect();
+
+        // /proc/net/route this is a system file which gives info about kernel's routing table and
+        // the default route has a destination of "00000000". First column gives Iface name.
+        if fields[1] == "00000000" {
```

Comment:
mmm it sounds strange that Clippy did not complain here, what if fields is empty?

Reply:
yeah , I tried using clippy::pedantic still it didnt complain!
anyways added a check if fields is empty it will go to next line.